src/common/dump.cpp

```diff
@@ -44,6 +44,39 @@ void dump_to_xflash(void) {
     w25x_fetch_error();
 }
 
+void dump_err_to_xflash(const char *error, const char *title) {
+    dumpmessage_t dump_message;
+    w25x_sector_erase(w25x_error_start_adress);
+    dump_message.not_displayed = 0; // validate message
+    strlcpy(dump_message.msg_title, title, sizeof(dump_message.msg_title));
+    strlcpy(dump_message.msg_content, error, sizeof(dump_message.msg_content));
+    w25x_program(w25x_error_start_adress, (uint8_t *)(&dump_message), sizeof(dumpmessage_t));
+    w25x_fetch_error();
+}
+
+int dump_in_xflash_get_message(char *msg_dst, uint16_t msg_dst_size, char *tit_dst, uint16_t tit_dst_size) {
+    dumpmessage_t dump_message;
+    w25x_rd_data(w25x_error_start_adress, (uint8_t *)(&dump_message), sizeof(dumpmessage_t));
+    if (w25x_fetch_error())
+        return 0;
+    strlcpy(tit_dst, dump_message.msg_title, sizeof(dump_message.msg_title));
+    strlcpy(msg_dst, dump_message.msg_content, sizeof(dump_message.msg_content));
+    return 1;
+}
+
+void dump_in_xflash_clear_message(void) {
+    w25x_sector_erase(w25x_error_start_adress);
```

Comment:
It looks like you are calling `w25x_sector_erase(w25x_error_start_adress)`
twice per error screen saved and displayed. This is probably lowering xFlash durability to half of what is achievable.

There are two possible solutions to that problem. 
1. Either you can read the whole sector, or at least `sizeof(dumpmessage_t)` in `dump_err_to_xflash()` and erase the sector only if it is already not erased.

or

2. You can implement what I already proposed earlier. Put two flags (`invalid`, `not_diplayed`) instead of just one (`not_diplayed`) into `dumpmessage_t` and switch them one by one. Make the message valid and not displayed in `dump_err_to_xflash()`. And make it displayed in `dump_in_xflash_clear_message()` without `w25x_sector_erase` call.